stonesoup/simulator/simple.py

```diff
@@ -60,15 +62,19 @@ class SwitchOneTargetGroundTruthSimulator(SingleTargetGroundTruthSimulator):
     The element in the ith row and the jth column is the probability of\
      switching from the ith transition model in :attr:`transition_models`\
      to the jth")
+    seed: Optional[int] = Property(default=None, doc="Seed for random number generation."
+                                                     " Default None")
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.index = 0
+        self.random_state = np.random.RandomState(self.seed)
 
     @property
-    def transition_model(self):
-        self.index = np.random.choice(range(0, len(self.transition_models)),
-                                      p=self.model_probs[self.index])
+    def transition_model(self, random_state=None):
```

Comment:
Can't have argument on _property_ like this, as no way to pass arguments. May be you could change this to `_transition_model` and add `transition_model = property(_transition_model)`, and then call it with `self._transition_model(random_state)` inside the simulation.

Maybe just remove `transition_model` altogether 🤷

Reply:
Now changed it to just use self.random_state. I think that should fix it.